src/odemis/acq/stream/_sync.py

```diff
@@ -1149,9 +1230,318 @@ def _runAcquisition(self, future):
         if hasattr(self, "useScanStage") and self.useScanStage.value:
             # TODO does not support polarimetry or image integration so far
             return self._runAcquisitionScanStage(future)
+        elif self._supports_hw_sync():
+            return self._runAcquisitionHwSyncEbeam(future)
         else:
             return self._runAcquisitionEbeam(future)
 
+    def _adjustHardwareSettingsHwSync(self):
+        """
+        Read the SEM and CCD stream settings and adapt the CCD and SEM scanner accordingly.
+        :returns: exp + readout (float): Estimated time for a whole (but not integrated) CCD image.
+                  integration_count (int): Number of images to integrate to match the requested exposure time.
+        """
+        if self._integrationTime and self._integrationCounts.value > 1:
+            # We would need to request the e-beam scanner to duplicate each pixel N times.
+            # (in order to send N triggers to the CCD)
+            raise NotImplementedError("Integration time not supported with hardware sync")
+
+        integration_count = 1
+
+        # The CCD has already been configured based on the SettingsStream settings, so no need to
```

Comment:
It's about the code that you *could* be expecting here, but it's not here because it's not needed... So it's not really related to any specific part. I've updated the comment to clarify it.
I've also found out that probably this is not completely true yet... so I need to fix it!

Reply:
I've extended the code to handle integrationTime.